### ISSUE TITLE
Make workflow within the plugin more strict

### ISSUE THREAD

**Issue by Kathrin84** [learnweb/moodle-mod_ratingallocate#78]
Hi, 
as we thought about a solution for the problem with the mapping issue of the group ids (-> Issue #33 ), we came up with another improvement of the workflow for the users.

We experienced that a user can change choices and also has the ability to re-create groups unlimited, even if the allocation already has been published.

We're asking ourselves if it wouldn't be better if the workflow would be a more strict one in which the not necessary buttons and options will be disabled.

Here's a flow chart to this issue:
![flussdiagramm](https://cloud.githubusercontent.com/assets/4711883/11441703/eabfd19e-950e-11e5-8504-f00191c805d3.png)

I'll explain a bit closer what this means.
The beginning is quite clear. The user adds the plugin to the course and makes his settings and creates the choices.
After the voting period is over, he runs the allocation algorithm. If he is not satisfied with the result, he can add or delete choices or do manual allocation.
Only if he ran the allocation the first time, the button for publishing will become active.

If he's satisfied with the allocation, he can publish it. 
When this button is hit, following changes:
* Choices are not editable anymore
* Allocation algorithm cannot be run anymore
* The button for creating groups become active.

If the user once has created the groups via the plugin, this button should also be disabled.
As the everything has been published, there's no need for any changes anymore.

As the groups won't need to be deleted for another creation instance, the group id's are not needed anymore.

What do you think about that proposal?

Cheers,
Kathrin

**Comment by tobiasreischmann** [comment]
Hey Kathrin,
I thought about this issue as well. However, I came up with beliefe that the current workflow even needs to be relaxed more. Two examples:

1. Consider the ability of the teacher to create groups:
- Some might prefer to create the groups when the allocation is done. This is of course needed to add the students to the course.
- However, some teachers might want to create the groups before the rating period even startet or during the rating period to be able to prepare group specific content in the course.

2. Consider the ability of the teacher to change the allocation after the publication:
- Students do not know about the allocation before it is published.
- There will always be students, who forgot about a different appointment and who will show up in the teachers office asking for a change of the allocation. Although there might be teachers who do not want to change the allocation in such cases, there might be some who want to.

For reaons like this I opened up the workflow to allow almost anything at anytime. However, I also added special hints for the teacher, which explain him in all different steps of the workflow, what he is supposed to do next and what he should do only with extrem caution, since it could end in results he has not indended (like the changing of choices after publishing the allocation)

I hope this is a way you are satisfied with.
Cheers,
Tobias
